library/blivet.py

```diff
@@ -231,7 +231,17 @@ def _resize(self):
         except Exception:
             raise BlivetAnsibleError("invalid size specification for volume '%s': '%s'" % (self._volume['name'], self._volume['size']))
 
-        if size and self._device.resizable and self._device.size != size:
+        if size and self._device.size != size:
+            try:
+                self._device.format.update_size_info()
+            except FSError:
+                raise BlivetAnsibleError("volume '%s' cannot be resize because it needs repair" % self._device.name)
```

Comment:
I think this isn't necessary, `update_size_info` shouldn't raise FSError anywhere now.

Reply:
@dwlehman @japokorn  should this comment be addressed?